opm/autodiff/SimulatorBase_impl.hpp

```diff
@@ -475,13 +318,47 @@ namespace Opm
         }
     } // namespace SimFIBODetails
 
-    template <class T>
-    void
-    SimulatorFullyImplicitBlackoil<T>::
-    Impl::computeRESV(const std::size_t               step,
-                      const Wells*                    wells,
-                      const BlackoilState&            x,
-                      WellStateFullyImplicitBlackoil& xw)
+    template <class Implementation>
+    void SimulatorBase<Implementation>::handleAdditionalWellInflow(SimulatorTimer& timer,
+                                                                   WellsManager& wells_manager,
+                                                                   WellState& well_state,
+                                                                   const Wells* wells)
+    { }
+
+    template <class Implementation>
+    auto SimulatorBase<Implementation>::createSolver(const Wells* wells)
+        -> std::shared_ptr<Solver>
```

Comment:
yep, it's supported. in 4.4 eWoms uses this in conjunction with `decltype` to implement perfect forwarding.